### PR TITLE
Fix missing omrsysinfo_get_x86_description on Windows

### PR DESCRIPTION
**Pull request by harryyu1994**
Signed-off-by: Harry Yu <harryyu1994@gmail.com>

### CODE REVIEW

File: port/common/omrsysinfo_helpers.c [hunk 1]
```diff
@@ -33,7 +33,7 @@
 
 #include <string.h>
 
-#if (defined(J9X86) || defined(J9HAMMER))
+#if (defined(OMR_OS_WINDOWS) || defined(J9X86) || defined(J9HAMMER))
```
Comment (rwy0717) [inline port/common/omrsysinfo_helpers.c]: while we're here, might as well remove the redundant parens:
```suggestion
#if defined(OMR_OS_WINDOWS) || defined(J9X86) || defined(J9HAMMER)
```

File: port/common/omrsysinfo_helpers.c [hunk 2]
```diff
@@ -224,4 +224,4 @@ omrsysinfo_get_x86_description(struct OMRPortLibrary *portLibrary, OMRProcessorD
 
 	return 0;
 }
-#endif /* (defined(J9X86) || defined(J9HAMMER)) */
+#endif /* defined(OMR_OS_WINDOWS) || (defined(J9X86) || defined(J9HAMMER)) */
```
Comment (rwy0717) [inline port/common/omrsysinfo_helpers.c]:
```suggestion
#endif /* defined(OMR_OS_WINDOWS) || (defined(J9X86) || defined(J9HAMMER) */
```

File: port/common/omrsysinfo_helpers.h [hunk 3]
```diff
@@ -30,9 +30,9 @@
 
 #include "omrport.h"
 
-#if (defined(J9X86) || defined(J9HAMMER))
+#if (defined(OMR_OS_WINDOWS) || defined(J9X86) || defined(J9HAMMER))
```
Comment (rwy0717) [inline port/common/omrsysinfo_helpers.h]:
```suggestion
#if defined(OMR_OS_WINDOWS) || defined(J9X86) || defined(J9HAMMER)
```

**Comment by rwy0717** [comment]
This line is probably also going to be problematic: https://github.com/eclipse/omr/pull/4803/files#diff-9cc0bc6336c8e4a04cfd5ab571dcc9a4R37

**Comment by harryyu1994** [comment]
> This line is probably also going to be problematic: https://github.com/eclipse/omr/pull/4803/files#diff-9cc0bc6336c8e4a04cfd5ab571dcc9a4R37

Why would be it problematic?

**Comment by rwy0717** [comment]
Sorry I thought I was looking at another place that needed fixing up, but you already fixed it.

**Comment by rwy0717** [comment]
@genie-omr build all

**Comment by rwy0717** [comment]
@genie-omr build osx

**Pull request merged**